client-js/app/scripts/activities.ts

```diff
@@ -112,14 +112,22 @@ export class DatavaultActivity implements Activity {
     };
   }
 
-  async onNewDir() {
-    var listing = await this.api.dir(this.path);
-    this.elem.dirs = this.getDirs(listing);
+  private onNewDir(item: {name: string, tags?: Array<string>}) {
+    const dir = {
+      name: item.name,
+      url: this.places.grapherUrl(this.path, item.name),
+      tags: (item.tags) ? item.tags : []
+    };
+    this.elem.newDir(dir);
```

Comment:
Could also omit `const` declarations here and in `onNewDataset` and just pass object literals to the appropriate functions.